flink-connector-pulsar/archunit-violations/f4d91193-72ba-4ce4-ad83-98f780dce581

```diff
@@ -10,3 +10,15 @@ org.apache.flink.connector.pulsar.source.PulsarSourceITCase does not satisfy: on
 * reside in a package 'org.apache.flink.runtime.*' and is annotated with @ExtendWith with class InternalMiniClusterExtension\
 * reside outside of package 'org.apache.flink.runtime.*' and is annotated with @ExtendWith with class MiniClusterExtension\
  or contain any fields that are public, static, and of type MiniClusterWithClientResource and final and annotated with @ClassRule or contain any fields that is of type MiniClusterWithClientResource and public and final and not static and annotated with @Rule
+org.apache.flink.connector.pulsar.sink.PulsarSinkITCase does not satisfy: only one of the following predicates match:\
```

Comment:
@syhily I'd like to know where this check comes from so that I can investigate to avoid further conflict.

That is - where is the source code of archunit violation check.

Reply:
Do you mean the source code here?

https://github.com/apache/flink/blob/3379c2c085da43bb452536c981a7fc13f39482ee/flink-architecture-tests/flink-architecture-tests-test/src/main/java/org/apache/flink/architecture/rules/ITCaseRules.java#L70-L140

Reply:
I guess the violation in PulsarSinkITCase caused by non static private final MiniClusterExtension.

Reply:
> I guess the violation in PulsarSinkITCase caused by non static field private final MiniClusterExtension.

Yes. But the actually nested test cases have. I don't think we should tweak to bypass the check..

Reply:
> I guess the violation in PulsarSinkITCase caused by non static field private final MiniClusterExtension.

After testing, I found that this guess is incorrect. The real reason is that the top level class must have a qualified mini cluster field and the nested classes are not considered.

Reply:
Maybe we can improve `ITCaseRules` in `apache/flink` repo to support our situation (Introducing `MiniCluster` only in nested classes). 🤔

Reply:
@reswqa Thanks for pointing out the source code! Perhaps I can take a look when I have free time (I do _not_ promise it, lol).

Do you think it's reasonable we duplicate a bit the exclusion to suppress failing daily workflows?

Reply:
The reason why we need to duplicate rather than directly modify is that our CI will support multiple versions of flink? 

Given that `FLINK-31804` has only been merged into the master(1.18), all external connectors that need to suppress this violation should also have the same problem.

Perhaps temporary duplicate is acceptable, as it will continue until the workflow for flink_version < 1.18 is no longer running.

Reply:
I thought about it again, and feel a bit that maybe `FLINK-31804` should be merged into 1.16 and 1.17. Because to some extent, this (without considering the `MiniClusterTestEnvironment`) is a bug for `ITCaseRules`. If it were backported to these versions, wouldn't there be no such duplicate. 🤔

Reply:
> I thought about it again, and feel a bit that maybe FLINK-31804 should be merged into 1.16 and 1.17. Because to some extent, this (without considering the MiniClusterTestEnvironment) is a bug for ITCaseRules. If it were backported to these versions, there is no need to duplicate it. 🤔 WDYT?

No need to duplicate. But still need to update the message.

I'd prefer to duplicate for now so that we can capture other failure in daily CI if any. We can always revert the duplicate when FLINK-31804 properly backport, instead of waiting here and letting the daily CI fails fast and conceal other possible issues.

Reply:
> I'd prefer to duplicate for now so that we can capture other failure in daily CI if any. We can always revert the duplicate when [FLINK-31804](https://issues.apache.org/jira/browse/FLINK-31804) properly backport, instead of waiting here and letting the daily CI fails fast and conceal other possible issues.

Yes, make sense.